### PR TITLE
client, cmd/snap: use the new multi-refresh endpoint

### PR DESCRIPTION
**Pull request by chipaca**
This builds on #1615 so [compare with that](https://github.com/chipaca/snappy/compare/refresh-all...chipaca:refresh-all-client).

multi-refresh does not currently offer a way to show errors beyond the catastrophic. That is, if the actual update fails it'll show an error, but if you ask it to update snaps you don't have installed, or for which there aren't updates, nothing is printed.

### CODE REVIEW

File: cmd/snap/cmd_snap_op.go [hunk 1]
```diff
@@ -325,36 +325,32 @@ type cmdRefresh struct {
...
 	cli := Client()
-	updates, _, err := cli.Find(&client.FindOptions{Refresh: true})
+	changeID, err := cli.RefreshMany(snaps, nil)
 	if err != nil {
-		return fmt.Errorf("cannot list updates: %s", err)
+		return err
 	}
-	// nothing to update/list
-	if len(updates) == 0 {
-		fmt.Fprintln(Stderr, i18n.G("All snaps up-to-date."))
```
Comment (niemeyer) [inline cmd/snap/cmd_snap_op.go]: What's happening to this case?  When do we report it?

File: cmd/snap/cmd_snap_op.go [hunk 2]
```diff
@@ -416,19 +412,19 @@ func (x *cmdRefresh) Execute([]string) error {
...
+		return refreshOne(x.Positional.Snaps[0], &client.SnapOptions{
+			Channel:  x.Channel,
+			DevMode:  x.DevMode,
+			JailMode: x.JailMode,
+		})
+	}
 
-		return refreshAll()
+	if x.asksForMode() || x.asksForChannel() {
+		return errors.New("a snap name is needed to specify mode or channel flags")
```
Comment (niemeyer) [inline cmd/snap/cmd_snap_op.go]: This error message doesn't seem to make sense if len(snaps) > 1.

File: cmd/snap/cmd_snap_op.go (hunk 1 above)
Comment (chipaca) [inline cmd/snap/cmd_snap_op.go]: down below :-)

File: client/snap_op.go [hunk 3]
```diff
@@ -76,15 +85,39 @@ func (client *Client) Revert(name string, options *SnapOptions) (changeID string
 func (client *Client) doSnapAction(actionName string, snapName string, options *SnapOptions) (changeID string, err error) {
 	action := actionData{
 		Action:      actionName,
-		Name:        snapName,
```
Comment (mvo5) [inline client/snap_op.go]: Nice elimination of redundancy!

File: daemon/api.go [hunk 4]
```diff
@@ -938,18 +942,64 @@ func isTrue(form *multipart.Form, key string) bool {
...
+	}
+	chg.Set("api-data", map[string]interface{}{"snap-names": updated})
+
+	return AsyncResponse(nil, &Meta{Change: chg.ID()})
+}
+
+func postSnaps(c *Command, r *http.Request, user *auth.UserState) Response {
 	contentType := r.Header.Get("Content-Type")
 
+	if contentType == "application/json" {
```
Comment (mvo5) [inline daemon/api.go]: Neat!

**Comment by mvo5** [comment]
Thanks, looks nice. Once we have multi operation install I would love to see a spread test too with `snap install hello-world test-snapd-tools`

**Pull request merged**